### ISSUE TITLE
Test result of .deb release on Ubuntu

### ISSUE THREAD

**Issue by blueskyson** [marchersimon/tldr-cpp#7]
I tried install via deb file but some error occurred, here is a brief log:

```
$ sudo dpkg -i tldr-cpp_1.6_amd64.deb
successful message here
$ tldr
tldr: error while loading shared libraries: libcurl.so.4: cannot open shared object file: No such file or directory
$ sudo apt install libcurl4-openssl-dev
$ tldr
tldr: error while loading shared libraries: libzip.so.4: cannot open shared object file: No such file or directory
$ sudo apt install libzip-dev
$ tldr
tldr: error while loading shared libraries: libzip.so.4: cannot open shared object file: No such file or directory
```

The `libzip.so.4` not found error still occurred after installing `libzip-dev`, the reason is that `libzip-dev` package depends on `libzip.so.5`, shown as below:

![Screenshot from 2021-09-04 21-07-54](https://user-images.githubusercontent.com/46962923/132095440-79ea59fa-c62d-48d2-b6a6-2c6968a4d2de.png)

Further more, libzip-dev-1.4 is not supported in Ubuntu 20.04's official packages according to [this](https://packages.ubuntu.com/search?suite=all&searchon=names&keywords=libzip).

**Comment by marchersimon** [comment]
I tested it on Ubuntu 21.04 and there I had no problems. It was a fresh install and the only packages I added were `libcurl4-openssl-dev` and  `libzip-dev`. But those were just for building. Even after I removed them, tldr still worked.

Would it be possible to add `libzip.so.4` as a dependency or is there anything I could to so it uses `libzip.so.5`?

**Comment by blueskyson** [comment]
Finally I found libzip4 package [here](https://debian.pkgs.org/sid/debian-main-amd64/libzip4_1.7.3-1_amd64.deb.html) and installed its deb from a mirror. After installing libzip4, Next error occurred:

```
$ tldr
tldr: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by tldr)
```

GLIBC_2.33 is required. I searched my glibc version, it's 2.31:

```
$ ldd --version
ldd (Ubuntu GLIBC 2.31-0ubuntu9.2) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```
According to another discussion, even the latest Debian 11 uses GLIBC_2.31 while Ubuntu 21.04 uses GLIBC_2.33

It's not suggest to force upgrade glibc because it causes **kernel panic** in high risk.

In conclusion, it's not recommended to install from this deb under Ubuntu 21.04 or all stable Debian.

**Comment by marchersimon** [comment]
I'll download Ubuntu 20.04 and compile it there from now on. I couldn't find a list of which distros use which version of glibc, but I hope most are compatible with 2.31. Thanks for the help so far.

**Comment by marchersimon** [comment]
I now managed to get it running under both versions. On Ubuntu 20.04 it runs out of the box and for Ubuntu 21.04, you have to link `libzip.so.5` to `libzip.so.4` first.

**Comment by blueskyson** [comment]
1.17 deb file works fine on my machine.
